app/models/passwordless/session.rb

```diff
@@ -55,10 +57,14 @@ def available?
     def set_defaults
       self.expires_at ||= Passwordless.expires_at.call
       self.timeout_at ||= Passwordless.timeout_at.call
-      self.token ||= loop {
-        token = Passwordless.token_generator.call(self)
-        break token unless Session.find_by(token: token)
-      }
+
+      if !token || !token_digest
+        self.token, self.token_digest = loop {
+          token = Passwordless.token_generator.call(self)
+          token_digest = Passwordless.digest(token)
+          break [token, token_digest] unless Session.find_by(token_digest: token_digest)
+        }
+      end
```

Comment:
This looks a bit suspicious to me. I don't think anything's wrong but it's hard to make sense of. Do we need/want to check for both `!token || !token_digest`. What if we have one but not the other? Will that ever happen? Could we prevent it? Just dumping my brain, sorry 😊